### ISSUE TITLE
Extract replay specific logic from DevToolsStartup

### ISSUE THREAD

**Issue by jasonLaster** [replayio/gecko-dev#396]
I personally find it hard to follow DevToolsStartup and understand how everything is related.

I think it might help if we first extracted replay specific logic and then began to think about the appropriate interface for some of these concerns:

1. button
2. starting/stopping tabs
3. handling user auth

Replay logic: https://gist.github.com/jasonLaster/cb6a0c13ec8a24aed24994e8df9908da
DevtoolsStartup.jsm https://gist.github.com/1e60f6cf21c96e6077c0e8398ef14359

**Comment by gideonred** [comment]
This ticket has been moved to [RUN-46](https://linear.app/replay/issue/RUN-46/extract-replay-specific-logic-from-devtoolsstartup)
